### DISCUSSION TITLE
Can we highlight block of text as single highlight?

### DISCUSSION THREAD

**Discussion by CR0405** [pymupdf/PyMuPDF#1055] [Q&A] [answered]
I'm trying to highlight multiple lines of text as single highlight by using addHighlightAnnot, but it's highlighting each line separately. 

For Example

Line No 1
Line No 2
Line No 3
I needed to get single rect value, but it's providing rect values for each line.

Is there is any way to get single rect value for all 3 lines.

**Comment by JorjMcKie** [comment]
First of all:
Why do you classify this as a bug? No to speak of the fact, that **_all_** mandatory information for a bug submission are missing ...
So I am taking the liberty to
1. reclassify this as a question
2. move your question to the "Discussions" tab using category "Looking for help". More responses can be found there.

**Comment by JorjMcKie** [comment, marked as answer]
Text marker annotations work on the basis of either single rectangles or quads (used in case of non-horizontal stuff), or on a list of such objects.

If a list is used, one text marker is created for all members of the list. But the highlighting (underlining, strikeout, etc.) is visually still the same as if you created separate text markers for each rectangle.

You are free to give any rectangle / quad to `page.addHighlightAnnot()` - there needs no text at all be contained in it: it could be empty or contain an image ... whatever.
So in your case simply join the line rectangles into one and use the result as argument for annot creation.
